File: txr/DesignPatternsChapter6/src/main/java/org/txr/designpatterns/chapter6/candidates/Politician.java
package org.txr.designpatterns.chapter6.candidates;

public class Politician {
	
	public void quote(String message) {
		System.out.println(this + ":\n " + message+"\n");
	}
	public String toString() {
		return this.getClass().getSimpleName();
	}
}

File: txr/DesignPatternsChapter6/src/main/scala/org/txr/designpatterns/chapter6/candidates/Politician.scala
package org.txr.designpatterns.chapter6.candidates

trait Politician {
  def quote(message:String): Unit = {
    System.out.println(this+":\n " + message + "\n")
  }
  override def toString(): String = {
    this.getClass().getSimpleName()
  }
}
